projects/packages/masterbar/tests/php/WPcom_Admin_Menu_Test.php

```diff
@@ -85,11 +85,16 @@ public function set_up() {
 
 		wp_set_current_user( static::$user_id );
 
-		$admin_menu = $this->getMockBuilder( WPcom_Admin_Menu::class )
-							->disableOriginalConstructor()
-							->onlyMethods( array( 'should_link_to_wp_admin' ) )
-							->getMock();
-		$admin_menu->method( 'should_link_to_wp_admin' )->willReturn( false );
+		$admin_menu = new class() extends WPcom_Admin_Menu {
+			// phpcs:ignore Generic.CodeAnalysis.UselessOverridingMethod.Found -- Not useless, changes visibility.
+			public function __construct() {
+				parent::__construct();
```

Comment:
The old code has `->disableOriginalConstructor()`. Wouldn't `parent::__construct();` theoretically change the behavior?

Also, why do we need to extend the `WPcom_Admin_Menu` class vs. creating a single-method class?

Reply:
This whole thing is super weird. As I mentioned in the commit message,

> This had a weird case. It creates a mock with `disableOriginalConstructor()`, but apparently relies on the inherited `::get_instance()` both `new`-ing a new instance and the constructor being called on the new instance.
>
> The `new`-ed instance breaks in PHPUnit 11 due to uninitialized fields. Trying to make the mock setup work for PHPUnit 11 broke it for 8 and 9.
>
> The simplest solution seems to be using an anonymous class instead, which doesn't have any phpunit state to not be initialized.
>
> P.S. The "Initialize in setUp so it registers hooks for every test" comment is wrong, it only registers the hooks on the first call because `::get_instance()` does singletons and nothing resets it.